### PR TITLE
drivers: pinctrl: max32: fix correct configuring drive strength

### PR DESCRIPTION
**Pull request by mertekren**
This commit fixes configuring pin drive strength in pinctrl driver. Previously, there was a mismatch while filling pincfg and checking pincfg drive strength field. This fix simplifies the operation and avoids gpio driver header dependency.

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
Please make commit title more clear

File: drivers/pinctrl/pinctrl_max32.c [hunk 1]
```diff
@@ -67,20 +66,7 @@ static int pinctrl_configure_pin(pinctrl_soc_pin_t soc_pin)
...
-		gpio_cfg.drvstr = MXC_GPIO_DRVSTR_2;
-		break;
-	case MAX32_GPIO_DRV_STRENGTH_3:
-		gpio_cfg.drvstr = MXC_GPIO_DRVSTR_3;
-		break;
-	default:
-		gpio_cfg.drvstr = MXC_GPIO_DRVSTR_0;
-		break;
-	}
+	gpio_cfg.drvstr = (pincfg >> MAX32_DRV_STRENGTH_SHIFT) & (0x3);
```
Comment (MaureenHelm) [inline drivers/pinctrl/pinctrl_max32.c:69]: You could use `FIELD_GET`

**Review by MaureenHelm — DISMISSED** [review]

File: drivers/pinctrl/pinctrl_max32.c (hunk 1 above)
Comment (ozersa) [inline drivers/pinctrl/pinctrl_max32.c:69]: Look goods, 
While updating commit message, please remove redundant parentheses from (0x3)

**Comment by mertekren** [comment]
> Please make commit title more clear

Done, tried to clarify a bit more.

**Review by gmarull — CHANGES_REQUESTED** [review]
commit title still misleading: `drivers: pinctrl: fix: correct configuring drive strength`, is it fixing all drivers?

File: drivers/pinctrl/pinctrl_max32.c (hunk 1 above)
Comment (mertekren) [reply drivers/pinctrl/pinctrl_max32.c:69]: 👎 the formatter checks okay, so i think keeping this way better if it's okay with you. Thanks

Comment (mertekren) [reply drivers/pinctrl/pinctrl_max32.c:69]: Hi Muareen thanks for suggestion. I checked this but there's a division there that i think didn't yield to same results
#define FIELD_GET(mask, value)  (((value) & (mask)) / LSB_GET(mask))

for example FIELD_GET(0x2, 0x777) gives 1, but in my case it should've given 2.

or maybe i'm missing something.

**Comment by mertekren** [comment]
> commit title still misleading: `drivers: pinctrl: fix: correct configuring drive strength`, is it fixing all drivers?

Done, added "max32"

**Review by gmarull — CHANGES_REQUESTED** [review]

**Comment by ttmut** [comment]
Compliance check fails due to a merge commit. Rebase instead.
https://github.com/zephyrproject-rtos/zephyr/actions/runs/11495235267/job/31994324544?pr=80173#step:7:27

File: drivers/pinctrl/pinctrl_max32.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (mertekren) [reply drivers/pinctrl/pinctrl_max32.c]: done.

**Comment by mertekren** [comment]
> Compliance check fails due to a merge commit. Rebase instead. https://github.com/zephyrproject-rtos/zephyr/actions/runs/11495235267/job/31994324544?pr=80173#step:7:27

Done

**Review by MaureenHelm — APPROVED** [review]

**Review by ozersa — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@gmarull can you take another look?

**Review by gmarull — APPROVED** [review]

**Pull request merged**